### PR TITLE
🔧 fix mint with bridged KSM

### LINKED ISSUE #5284: Not possible to mint with only bridged KSM
### What happened?

I wanted to create a new collection on Basilisk with my new wallet. I sent KSM on this wallet. Bridged it to Basilisk chain. 
I can see my bridged KSM in my bsx wallet. Fees are in KSM.

Screenshot 2023-03-16 at 17 37 58

Screenshot 2023-03-16 at 17 38 49

I'm unable to create a collection. Insufficient funds. Even though I have enough funds to mint. From what I know minting should work without the need to have BSX tokens in a wallet - especially if my fees are set to KSM.
Screenshot 2023-03-16 at 17 36 31

### Please reproduce in steps

Click on create - insufficient funds

### Expected Behavior

Mint without bsx tokens

### What browsers are you seeing the problem on?

Chrome

### At which address did you encounter bug?

kodadot.xyz

### Are you logged in?

Yes

### Which wallet you are using?

PolkadotJS

### Screenshots

-

### Relevant log output

_No response_

### Payment link for reward

https://kodadot.xyz/transfer/?target=%3CMy_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address%3E

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #5306: Unable to create collection on Basilisk with Talisman
### What happened?

When trying to create a collectio on BSX a warning shows that the user has insufficient funds, even though the user has the necessary funds.

### Please reproduce in steps

Create a collection connect the Talisman wallet with the necessary funds, create a collection on Basilisk and click create collection. 

### Expected Behavior

Expected behaviour would be that a collection is minted.

### What browsers are you seeing the problem on?

Desktop Chrome

### At which address did you encounter bug?

beta.kodadot.xyz, kodadot.xyz

### Are you logged in?

Yes

### Which wallet you are using?

Talisman

### At which chain did you encounter bug?

Basilisk

### Screenshots

Screenshot 2023-03-20 at 12 28 31

### Relevant log output

_No response_

### Payment link for reward

https://kodadot.xyz/transfer/?target=%3CMy_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address%3E

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #5284
- [x]  Closes #5306 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image
image

### CODE REVIEW

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Comment by daiagi** [comment]
doesn't work if paying fees in BSX
changing to draft

**Comment by daiagi** [comment]
@JustLuuuu  does it work for you?

**Comment by daiagi** [comment]
@kodadot/internal-dev I think the deposit amount should be displayed in the chosen fee token.
Wdyt?

File: components/bsx/Create/CreateCollection.vue [hunk 1]
```diff
@@ -101,6 +104,38 @@ export default class CreateCollection extends mixins(
...
+  }
+
+  @Watch('accountId', { immediate: true })
+  onAccountIdChange(val: string, oldVal: string) {
+    if (shouldUpdate(val, oldVal)) {
+      this.getFeesToken()
+    }
+  }
+
+  KsmToBsx(KsmValue: number) {
```
Comment (vikiival) [inline components/bsx/Create/CreateCollection.vue:129]: please follow eslint rules: function and parameter names starts with small letter

**Review by vikiival — CHANGES_REQUESTED** [review]
please move the functions somewhere else and do not duplicate code

File: components/bsx/Create/CreateCollection.vue (hunk 1 above)
Comment (daiagi) [reply components/bsx/Create/CreateCollection.vue:129]: sure thing. my bad

**Comment by daiagi** [comment]
> please move the functions somewhere else and do not duplicate code

@vikiival 
Done, please have a look again

File: components/bsx/Create/utils.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { useFiatStore } from '@/stores/fiat'
+import { getAssetIdByAccount } from '@/utils/api/bsx/query'
+
+export type Token = 'BSX' | 'KSM'
+
+export const ksmToBsx = (KsmValue: number): number => {
```
Comment (vikiival) [inline components/bsx/Create/utils.ts:11]:
```suggestion
export const ksmToBsx = (value: number): number => {
```

File: components/bsx/Create/utils.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  const fiatStore = useFiatStore()
+  const KSMToUsd = fiatStore.getCurrentKSMValue
+  const BSXToUsd = fiatStore.getCurrentBSXValue
+  return KsmValue * (Number(KSMToUsd) / Number(BSXToUsd))
```
Comment (vikiival) [inline components/bsx/Create/utils.ts:15]:
```suggestion
  return value * (Number(KSMToUsd) / Number(BSXToUsd))
```

File: components/bsx/Create/utils.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+export const getBalance = (token?: Token): number => {
+  const { balance } = useAuth()
+  const { $store } = useNuxtApp()
+
+  if (token === 'KSM') {
+    return $store.getters.getTokenBalanceOf('1')
+  }
+  return balance.value
+}
+
```
Comment (vikiival) [inline components/bsx/Create/utils.ts]: this wont work on snek

Comment (vikiival) [reply components/bsx/Create/utils.ts]: Correct ids for 
- snek is `5`
- bsx is `1`

**Review by vikiival — CHANGES_REQUESTED** [review]

**Comment by daiagi** [comment]
> it's mergeable?

I think yes

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $35 @ 34.41 USD/KSM ~ 1.017 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0x90c23a695b4301c450c5473a532372770c731ddd33d7dbc192fae46e3a531192](https://kusama.subscan.io/extrinsic/0x90c23a695b4301c450c5473a532372770c731ddd33d7dbc192fae46e3a531192)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**